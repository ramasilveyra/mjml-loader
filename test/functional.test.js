import path from 'path';
import del from 'del';
import webpack from 'webpack';

// TODO: For some reason jest in watch mode loops infinitely in this test.
// Needs investigation, skiped by default.
describe.skip('while using mjml-loader with raw-loader', () => {
  beforeEach(() => del(path.resolve(__dirname, 'fixtures/bundle.js')));

  it('should inline MJML result into the bundle', async () => {
    const options = {
      entry: path.resolve(__dirname, 'fixtures/entry.js'),
      output: {
        path: path.resolve(__dirname, 'fixtures'),
        filename: 'bundle.js',
        libraryTarget: 'commonjs2'
      },
      module: {
        rules: [{ test: /\.mjml$/, loader: require.resolve('../') }]
      },
      target: 'node'
    };

    await compile(options);
    const html = (await import('./fixtures/bundle')).default;

    expect(html).toMatchSnapshot();
  });
});

function compile(options) {
  return new Promise((resolve, reject) => {
    const compiler = webpack(options);
    compiler.run((err, stats) => {
      if (err || stats.hasErrors()) {
        return reject(err || stats);
      }

      return resolve(stats);
    });
  });
}
