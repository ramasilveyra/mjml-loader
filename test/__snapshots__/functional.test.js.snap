// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using mjml-loader with raw-loader should inline MJML result into the bundle 1`] = `
"<!doctype html>
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
<head>
  <title></title>
  <!--[if !mso]><!-- -->
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
  <!--<![endif]-->
<meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
<style type=\\"text/css\\">
  #outlook a { padding: 0; }
  .ReadMsgBody { width: 100%; }
  .ExternalClass { width: 100%; }
  .ExternalClass * { line-height:100%; }
  body { margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; }
  table, td { border-collapse:collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt; }
  img { border: 0; height: auto; line-height: 100%; outline: none; text-decoration: none; -ms-interpolation-mode: bicubic; }
  p { display: block; margin: 13px 0; }
</style>
<!--[if !mso]><!-->
<style type=\\"text/css\\">
  @media only screen and (max-width:480px) {
    @-ms-viewport { width:320px; }
    @viewport { width:320px; }
  }
</style>
<!--<![endif]-->
<!--[if mso]>
<xml>
  <o:OfficeDocumentSettings>
    <o:AllowPNG/>
    <o:PixelsPerInch>96</o:PixelsPerInch>
  </o:OfficeDocumentSettings>
</xml>
<![endif]-->
<!--[if lte mso 11]>
<style type=\\"text/css\\">
  .outlook-group-fix {
    width:100% !important;
  }
</style>
<![endif]-->

<!--[if !mso]><!-->
    <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
    <style type=\\"text/css\\">

        @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);

    </style>
  <!--<![endif]--></head>
<body>
  
  <div style=\\"cursor:auto;color:#000000;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:22px;text-align:left;\\">Hello World!</div>
</body>
</html>"
`;
